### PR TITLE
add report for all versions

### LINKED ISSUE #4509: report: all version descriptions for all versions of all druids
Not sure if this is possible or possible but very difficult, but I would like a report of all version descriptions for all versions of all druids. I think these four fields in a CSV or TSV would be sufficient:

- druid
- version number (as a whole number)
- tag (or whatever the pseudo-semantic version number is called, e.g. 2.1.0
- version description

For various threads of SDR planning, especially around versioning, I would like to do some analysis on existing data. I'm especially interested in how many versions are:

- clearly system-described (usually via bulk action)
- are metadata bulk updates, which usually get a system-made description
- were obvious fixes to errors ("fix version mismatch")
- show some indication of user attempts to signify that some versions are more significant than others (i.e. "release 2.0")

### PR DESCRIPTION
**Pull request by peetucket**
## Why was this change made? 🤔

Fixes #4509 - Add a report for exporting all version info metadata to CSV file

## How was this change tested? 🤨

By running this branch on sdr-infra

### CODE REVIEW

File: app/reports/version_list.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    output_file = 'tmp/version_list_report.csv'
+
+    CSV.open(output_file, 'w') do |csv|
+      csv << %w[druid version tag description]
+
+      num_versions = ObjectVersion.all.count
+      puts "Found #{num_versions} object versions"
+      i = 0
+      ObjectVersion.find_each do |object_version|
+        i += 1
```
Comment (peetucket) [inline app/reports/version_list.rb:20]: the `find_each` ensures we do this in batches (1000 by default) so memory consumption is reasonable

File: app/reports/version_list.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+require 'csv'
```
Comment (mjgiarlo) [inline app/reports/version_list.rb:3]: Did you find you needed this or is it copypasta? I'd be surprised if we need this manual require. Rails should do this automatically.

File: app/reports/version_list.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Invoke via:
+# bin/rails r -e production "VersionList.report"
+class VersionList
+  def self.report
+    output_file = 'tmp/version_list_report.csv'
+
+    CSV.open(output_file, 'w') do |csv|
+      csv << %w[druid version tag description]
+
+      num_versions = ObjectVersion.all.count
```
Comment (mjgiarlo) [inline app/reports/version_list.rb:16]: I don't think the `.all` is needed here.

File: app/reports/version_list.rb (hunk 2 above)
Comment (peetucket) [reply app/reports/version_list.rb]: Removed and still works.

File: app/reports/version_list.rb (hunk 3 above)
Comment (peetucket) [reply app/reports/version_list.rb]: Removed

**Review by mjgiarlo — APPROVED** [review]
Thanks for considering my review!

**Pull request merged**